File: languages/csharp/SecureMemory/SecureMemory.Tests/ProtectedMemoryImpl/MemoryLimitExceptionTest.cs
using GoDaddy.Asherah.SecureMemory.ProtectedMemoryImpl;
using Xunit;

namespace GoDaddy.Asherah.SecureMemory.Tests.ProtectedMemoryImpl
{
    [Collection("Logger Fixture collection")]
    public class MemoryLimitExceptionTest
    {
        private const string Message = "Failure message";

        [Fact]
        private void MemoryLimitExceptionConstructorTest()
        {
            MemoryLimitException exception = new MemoryLimitException(Message);
            Assert.Equal(Message, exception.Message);
        }
    }
}

File: java/secure-memory/src/test/java/com/godaddy/asherah/securememory/protectedmemoryimpl/MemoryLimitExceptionTest.java
package com.godaddy.asherah.securememory.protectedmemoryimpl;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MemoryLimitExceptionTest {
  static String message = "Test Message";

  @Test
  void memoryLimitExceptionConstructorTest() {
    MemoryLimitException ex = new MemoryLimitException(message);
    assertEquals(message, ex.getMessage());
  }
}
